### PR TITLE
Json-editable menus

### PR DESCRIPTION
**Pull request by Dysfunk**
I made the main menu editable via jsons, in images/menuJson.json, you can change the x position of the menu and the angle of the list elements

(I am very new to haxe and this is my first ever pr btw, so if there is any problems feel free to tell me)

### CODE REVIEW

**Review by DM-kun — APPROVED** [review]
I approve this quality of life feature

**Review by ZAngryCadet — APPROVED** [review]

**Review by MasterDirector99 — APPROVED** [review]
**I APPROVED THIS COMMIT!!!**

**Review by HiroMizuki — APPROVED** [review]
ok, maybe???????????????

**Review by crowplexus — APPROVED** [review]
perfect, just perfect.

File: source/MainMenuState.hx [hunk 1]
```diff
@@ -136,9 +137,9 @@ class MainMenuState extends MusicBeatState
 		for (i in 0...optionShit.length)
 		{
 			var offset:Float = 108 - (Math.max(optionShit.length, 4) - 4) * 80;
-			var menuItem:FlxSprite = new FlxSprite(menuJSON.menulocation, (i * 140)  + offset);
-			menuItem.scale.x = scale;
-			menuItem.scale.y = scale;
+			var menuItem:FlxSprite = new FlxSprite(, (i * 140)  + offset);
```
Comment (Gui-iago) [inline source/MainMenuState.hx:140]: this prevents compiling

**Comment by Gui-iago** [comment]
nvm u fixed it, just ignore lol

**Comment by min89xd** [comment]
is perfect

**Comment by BeansCool54** [comment]
what a chad

**Review by FuseIsHere813 — APPROVED** [review]

**Comment by bobNewYear** [comment]
OK

**Review by Jotaro-Gaming — APPROVED** [review]
this is useless

**Comment by HafimieForever** [comment]
bruh why are you so lazy to build the game urself lol

**Comment by HiroMizuki** [comment]
> bruh why are you so lazy to build the game urself lol

wtf what are u talking about

**Comment by tposejank** [comment]
> bruh why are you so lazy to build the game urself lol

true

**Comment by tposejank** [comment]
the people that wish they had this are just lazy asses LMFAO

**Comment by PlugBoyOfficial** [comment]
> 

get over yourself dude. knowing how to program in haxe doesnt entitle you to a large ego

**Comment by tposejank** [comment]
> > 
> 
> get over yourself dude. knowing how to program in haxe doesnt entitle you to a large ego

at this point the fucking engine will become json based. the game wont even run on source code its just fucking jsons.

and making it in haxe is easier plus i dont see anyone would actually use this

**Comment by HiroMizuki** [comment]
> > get over yourself dude. knowing how to program in haxe doesnt entitle you to a large ego
> 
> at this point the fucking engine will become json based. the game wont even run on source code its just fucking jsons.
> 
> and making it in haxe is easier plus i dont see anyone would actually use this

yeah, i think you're right

**Comment by MrClogsworthYT** [comment]
> > > 
> > 
> > 
> > get over yourself dude. knowing how to program in haxe doesnt entitle you to a large ego
> 
> at this point the fucking engine will become json based. the game wont even run on source code its just fucking jsons.
> 
> and making it in haxe is easier plus i dont see anyone would actually use this

true lmfao

**Comment by Steviegt6** [comment]
> > > 
> > 
> > 
> > get over yourself dude. knowing how to program in haxe doesnt entitle you to a large ego
> 
> at this point the fucking engine will become json based. the game wont even run on source code its just fucking jsons.
> 
> and making it in haxe is easier plus i dont see anyone would actually use this

I see no reason why it's bad to push a data-oriented paradigm in something like this. JSONs are entirely practical for this sort of thing. If you want to use a source build, it's still entirely within your ability regardless of JSON implementations, is it not?

**Comment by MrClogsworthYT** [comment]
> > > > 
> > > 
> > > 
> > > get over yourself dude. knowing how to program in haxe doesnt entitle you to a large ego
> > 
> > 
> > at this point the fucking engine will become json based. the game wont even run on source code its just fucking jsons.
> > and making it in haxe is easier plus i dont see anyone would actually use this
> 
> I see no reason why it's bad to push a data-oriented paradigm in something like this. JSONs are entirely practical for this sort of thing. If you want to use a source build, it's still entirely within your ability regardless of JSON implementations, is it not?
> 
> I can sympathize with your latter comment, though. I don't see much use for this, specifically, given the purpose of Lua modding.
> 
> EDIT: Was writing under the belief this gave modder a way to add their own menu items. That is not the case. This is literally useless.

same i approved thinkin it was gonna make lua state shit or something

**Comment by TerrorGaming473** [comment]
can i use this in my engine? (I'll give the credits)

**Comment by Dysfunk** [comment]
> can i use this in my engine? (I'll give the credits)

You are free to do so

**Comment by Lokitot** [comment]
showcase video?

**Pull request closed without merging**